src/main.rs

```diff
@@ -1,9 +1,16 @@
+#![deny(rust_2018_idioms)]
+#![deny(non_upper_case_globals)]
+#![deny(non_camel_case_types)]
+#![deny(non_snake_case)]
+#![deny(unused_mut)]
+
```

Comment:
`deny(missing_docs)` removed here as `main.rs` is kind of special, things are almost clear with well doc-ed backend api.